<a href="https://colab.research.google.com/github/Hanizakkk/flyrank_working-repo/blob/main/Copy_of_w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Hanizakkk/flyrank_working-repo/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why
Content refresh and search performance. I want to investigate which pages may be worth reviewing or refreshing based on their search performance and content characteristics. I chose this lane because the dataset contains information about search visibility, click-through rate, page position, content age, and traffic-related metrics, which gives me several signals to explore.

In [ ]:

%cd /content/flyrank_working-repo

/content/flyrank_working-repo


In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))
print("Declining pages:", (df["trend_direction"] == "down").sum())
print("Growing pages:", (df["trend_direction"] == "up").sum())

Total pages: 30000
Declining pages: 16262
Growing pages: 4388


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*### The question

How can we identify which pages should be prioritized for content refresh?

The decision is whether a page should be reviewed or refreshed first. An SEO or content team could use the results to prioritize its work. A wrong recommendation could waste time and resources on pages that do not need changes, while pages that need attention may be missed.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
print("Search volume vs impressions correlation:",
      round(df["search_volume"].corr(df["impressions_90d"]), 3))

print("Pages:", len(df))

print("Median word count by trend:")
print(df.groupby("trend_direction")["word_count"].median())

Search volume vs impressions correlation: 0.001
Pages: 30000
Median word count by trend:
trend_direction
down      2909.0
flat      2698.5
new       2239.0
stable    2912.5
up        2847.5
Name: word_count, dtype: float64


## 3. Quick look at the data (2-3 real numbers)
### Quick look at the data

The dataset contains 30,000 pages, of which 16,262 are currently classified as declining and 4,388 as growing.

The correlation between search volume and 90-day impressions is 0.001, which is very close to zero. This suggests that search volume alone is not strongly related to the impressions a page actually receives in this dataset.

These observations suggest that multiple signals may be useful when deciding which pages should be prioritized for review.
*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
print("Total pages:", len(df))

print("Declining pages:",
      (df["trend_direction"] == "down").sum())

print("Growing pages:",
      (df["trend_direction"] == "up").sum())

print("Search volume vs impressions correlation:",
      round(df["search_volume"].corr(df["impressions_90d"]), 3))

Total pages: 30000
Declining pages: 16262
Growing pages: 4388
Search volume vs impressions correlation: 0.001


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*### What I can and can't claim

My analysis can identify observed and measured patterns in the available dataset. The results can provide directional insights and decision-support for prioritizing pages for review.

I cannot claim that the analysis proves that one factor causes a page to perform better, and I cannot claim that the model predicts or explains Google's ranking algorithm.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
print("Claims are limited to observed, measured, directional patterns and decision-support.")
print("No causal claims or claims about predicting Google's algorithm.")

Claims are limited to observed, measured, directional patterns and decision-support.
No causal claims or claims about predicting Google's algorithm.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.